smart_control/simulator/building_radiation_utils.py

```diff
@@ -0,0 +1,841 @@
+"""Building Radiation Utility Functions
+
+For computing the physical and thermal characteristics of buildings.
+"""
+
+from collections import deque
+import math
+from typing import Optional, Tuple
+
+import numpy as np
+
+from smart_control.simulator import constants
+
+# we are choosing to keep the mathematical notation names for the functions and
+# variables in this file
+# pylint: disable=invalid-name
+
+
+def calculate_A_tilde_inv(epsilon: np.ndarray, F: np.ndarray) -> np.ndarray:
+  """Calculates the A-tilde matrix used in radiative heat transfer calculations.
+
+  The A-tilde matrix relates the radiosity to the blackbody emissive power in a
+  radiative heat transfer system. It accounts for both emission and reflection.
+
+  Args:
+      epsilon: Array of surface emissivity values (between 0 and 1)
+      F: View factor matrix
+
+  Returns:
+      The A-tilde matrix relating radiosity to blackbody emissive power
+
+  Raises:
+      AssertionError: If emissivity vector size doesn't match view factor matrix
+          or if emissivity values are outside [0,1]
+  """
+  n = epsilon.shape[0]
+  epsilon[epsilon == 0] = 1e-10
+
+  A = np.eye(n)
+  I = np.eye(n)
+  for i in range(n):
+    for j in range(n):
+      A[i, j] = (I[i, j] - (1 - epsilon[i]) * F[i, j]) / epsilon[i]
+  return np.linalg.inv(A)
+
+
+def calculate_IFAinv(F: np.ndarray, A_inv: np.ndarray) -> np.ndarray:
+  r"""
+  Calculates the $IFA_{inv}$ matrix.
+
+  $$IFA_{inv} = (I - F) \tilde{A}^{-1}$$
```

Comment:
FYI - equation as displayed in the documentation site:

<img width="1478" height="543" alt="Image" src="https://github.com/user-attachments/assets/644ce2b6-a63b-4fda-b5a2-3362adb5b852" />